python/cudf/cudf/core/column/categorical.py

```diff
@@ -962,18 +1083,23 @@ def unique(self):
             ordered=self.ordered,
         )
 
-    def _encode(self, value):
+    def _encode(self, value) -> "ScalarObj":
         return self.categories.find_first_value(value)
 
-    def _decode(self, value):
+    def _decode(self, value: int) -> "ScalarObj":
         if value == self.default_na_value():
             return None
         return self.categories.element_indexing(value)
 
-    def default_na_value(self):
+    def default_na_value(self) -> "ScalarObj":
         return -1
 
-    def find_and_replace(self, to_replace, replacement, all_nan):
+    def find_and_replace(
+        self,
+        to_replace: Union["ColumnBase", list],
+        replacement: Union["ColumnBase", list],
```

Comment:
Is this generic enough to capture how these APIs are used? What if someone passes a pd.Series or a numpy array?

Reply:
Used ColumnLike isntead